### DISCUSSION TITLE
Repeated SSH Key Prompts When Connecting via SFTP with Both SSH Key and Password Authentication

### DISCUSSION THREAD

**Discussion by dlindegaard** [iterate-ch/cyberduck#16445] [Q&A]
**Describe the bug** 
When connecting to a server via SFTP using both an SSH key and a password, Cyberduck continuously prompts for the SSH key. This issue does not occur when connecting to servers requiring only password or SSH key authentication.

**To Reproduce Steps** 
1. Open Cyberduck on macOS.
2. Click "Open Connection" and select SFTP (SSH File Transfer Protocol).
3. Enter the server address and username.
4. Under "SSH Private Key", select your SSH key file.
5. Enter the password for your user account.
6. Click "Connect".
7. Observe that Cyberduck continuously prompts for the SSH key.

Expected behavior After providing the SSH key and password once, Cyberduck should connect to the server without repeatedly prompting for the SSH key.

Desktop:

OS: macOS Sonoma 14.5
Cyberduck Version: 9.0.2

**Additional context** 
This issue occurs only when connecting to servers requiring both SSH key and password authentication. It does not happen with servers that need only one authentication method.

**Comment by dkocher** [comment]
Can you share a [debug log](https://docs.cyberduck.io/cyberduck/support/#logging-output) and your SSH server configuration specifying authentication requirements if possible.

**Reply by dkocher** [reply]
I have opened #16895.

**Comment by dlindegaard** [comment]
@dkocher I checked the debug log, and no logging occurs about this

**Reply by dkocher** [reply]
Can you confirm the SSH private key you select is outside the `~/.ssh` folder.

**Reply by gulgiraf** [reply]
@dkocher It is within the ~/.ssh folder

**Reply by chlangley** [reply]
Did this get resolved?  If so please point me in that direction.

I am attempting to open sftp connections without password or username (based on KEYS as I do at the CLI) .  But I have not found a way to get Cyberduck to follow suit.  Again, if this is already documented please point me in the right direction.
